config/v1/types_infrastructure.go

```diff
@@ -52,10 +52,22 @@ type InfrastructureStatus struct {
 	// For more info: https://github.com/etcd-io/etcd/blob/329be66e8b3f9e2e6af83c123ff89297e49ebd15/Documentation/op-guide/clustering.md#dns-discovery
 	EtcdDiscoveryDomain string `json:"etcdDiscoveryDomain"`
 
-	// apiServerURL is a valid URL with scheme(http/https), address and port.
-	// apiServerURL can be used by components like kubelet on machines, to contact the `apisever`
-	// using the infrastructure provider rather than the kubernetes networking.
-	APIServerURL string `json:"apiServerURL"`
+	// apiServerURL is a URL for the Kubernetes API.
+	//
+	// Deprecated: Use apiServerExternalURI or apiServerInternalURI instead.
+	// +optional
+	APIServerURL string `json:"apiServerURL,omitempty"`
+
+	// apiServerExternalURI is a valid URI with scheme(http/https),
+	// address and port.  apiServerExternalURI can be used by components
+	// like the web console to tell users where to find the Kubernetes API.
+	APIServerExternalURI string `json:"apiServerExternalURI"`
```

Comment:
Calling it a URI instead of a URL might cause confusion. Is there a reason for the distinction?

Reply:
We use URL in most places, be consistent.

Reply:
Rerolled in b47d12f9 -> a33660b8 to stick with URL.